sim/pokemon.ts

```diff
@@ -192,6 +192,7 @@ export class Pokemon {
 	isStarted: boolean;
 	duringMove: boolean;
 
+	weighthg: number;
```

Comment:
Oh, I see, we're caching this value? Should this be cached, though?

Reply:
I believe this more accurately represents the way Autotomize affects weight in Gen 7; while in Gen 6 it was a weight modification effect which persisted across forme changes, in Gen 7 it applies as if it's property of the current forme, much like the type.

Reply:
Weight also resets on forme changes in Generation 6. An Aggron that used Autotomize twice in pre-Mega can be picked up by Sky Drop, but Mega Evolving causes Mega Aggron to retain the Speed boosts, yet not be picked up by Sky Drop, because its weight reset.

Reply:
We should maybe add a test for that. This also should maybe not be in the same commit, or at least be documented in the commit summary if it's a lot of work to pull them apart.

Reply:
I know your PR description says "implement the mechanics described here:", but I'd rather you explicitly list which mechanics you implement.

Reply:
Autotomize also stacks in Generation 5. Weight does not reset on forme changes in Generation 5.

> Generation 5: a Shaymin-Sky that uses Autotomize, then is frozen to revert to Land Forme Shaymin, gets hit with a Heavy Slam from Tynamo.  It deals 65 damage to a level 100, 229 Attack move against the 213 Defense Shaymin. This implies Heavy Slam was 80 BP rather than 40 BP, notably implying that weight does not reset on forme changes in this generation (Shaymin is the only candidate anyway, which also can't learn Autotomize in any case naturally).

Autotomize also does stack in Generation 5. After 3 Autotomizes, a level 17 Timburr with 35 Attack dealt 16 damage with Low Kick to a 185 Defense Aggron, which implies Low Kick was 80 HP, which would not be possible if only one Autotomize were being applied.

Reply:
- Updated for SSB changes
- Moved Autotomize weight not resetting to Gen 5
- Some tests already exist in `test/sim/misc/weight.js`, do you need others?

Actual alterations to mechanics implemented:
- Weight is calculated in positive integer multiples of hectograms, in particular for Light Metal
- Heavy Metal always applies before Float Stone
- Since Gen 6, weight is a property of the forme that is directly modified by Autotomize